odoo/addons/base/models/ir_attachment.py

```diff
@@ -83,6 +83,11 @@ def force_storage(self):
     def _migrate(self):
         record_count = len(self)
         storage = self._storage().upper()
+        # When migrating to filestore verifying if the directory has write permission
+        if storage == 'FILE':
+            filestore = self._filestore()
+            if not os.access(filestore, os.W_OK):
+                raise PermissionError(f"Write permission denied for filestore directory: {filestore}")
```

Comment:
Please don't put the filestore directory in the error message.  That's a potential security issue.  The message should simply say "Write permission denied for filestore."

Reply:
ok got it